subsys/net/ip/net_context.c

```diff
@@ -1246,6 +1246,14 @@ NET_CONN_CB(tcp_established)
 		return NET_DROP;
 	}
 
+	/* Increment the ack */
```

Comment:
This is *not* a correct change. First, application gets a chance to process incoming data. An application may reject or fail to process data (e.g., due to lack of memory). ACK gets sent only after that, and includes outcomes of the processing status (e.g., updated/not updated receive window, etc.).

If you look thru the history, you'll likely find that the opposite change was done, which you now try to revert.